flux-typeck/src/checker.rs

```diff
@@ -887,6 +879,24 @@ impl<'a, 'tcx, P: Phase> Checker<'a, 'tcx, P> {
     }
 }
 
+/// `init_constr` updates the `RefineCtxt` and `TypeEnv` with
+/// the bindings for a single `Constraint` which corresponds to
+/// - an input parameter of the function (from a .requires) or
+/// - a global type binding from a global constant
+
```

Comment:
I think this comment is outdated as it doesn't deal with consts at all. maybe we can just put the code back where it was.